File: java/src/entity/Product.java
package entity;

public class Product {
    private String name;
    private float price;
    private int stock;

    public Product(String name, float price, int stock) {
        this.name = name;
        this.price = price;
        this.stock = stock;
    }

    public String getName() {
        return name;
    }

    public float getPrice() {
        return price;
    }

    public int getStock() {
        return stock;
    }

    public void decreaseStock(int quantity) {
        if (quantity <= 0) {
            throw new IllegalArgumentException("La quantite doit etre superieure a 0");
        }

        if (this.stock < quantity) {
            throw new RuntimeException("Stock insuffisant pour " + this.name);
        }

        this.stock -= quantity;
    }
}

File: php/entity/Product.php
<?php

namespace entity;

class Product
{
    public function __construct(
        private string $name,
        private float $price,
        private int $stock
    ) {}

    public function getName(): string
    {
        return $this->name;
    }

    public function getPrice(): float
    {
        return $this->price;
    }

    public function getStock(): int
    {
        return $this->stock;
    }

    public function decreaseStock(int $quantity): void
    {
        if ($quantity <= 0) {
            throw new \InvalidArgumentException("La quantité doit être supérieure à 0");
        }

        if ($this->stock < $quantity) {
            throw new \RuntimeException("Stock insuffisant pour {$this->name}");
        }

        $this->stock -= $quantity;
    }
}